### ISSUE TITLE
[TT 1444] Machines and MultiVerse

### ISSUE THREAD

**Issue by fiddlycraft** [MinimallyCorrect/TickThreading#388]
It seems that the last working version that doesn't mess up machines is 1416 and I have been using that waiting for a fix with the MV and IC2 issues.
Please get this fixed.

**Comment by LunNova** [comment]
In what way are they "messed up"?

**Comment by alicheek** [comment]
what I can vouch to this is that in the multiverse world the IC machines don't work they now face the right way when placed but u cant right click them :) haven't tested other items yet (I don't really go in that world )

**Comment by LunNova** [comment]
Looks like the packet manipulation I was trying to do to fix dimension IDs in IC2's network manager doesn't work correctly, will look into it.

The change was necessary, as the way it used to work resulted in multiverse dimensions having the same IDs as non-multiverse dimensions, causing issues ranging from golems trying to visit blocks in other worlds to crashing the server under some circumstances. Of course, the dimension IDs would still need to be sent the old way to the client, or it just crashes, so that means that TT needs to rewrite any packets sent by mods containing the dimension ID to send the dimension ID the client thinks it is in. This shouldn't be necessary, but IC2 pointlessly implemented their own networking system. :/

**Comment by alicheek** [comment]
Yea I noticed when you changed it. Seemed much more stable when  u sorted it, great job as always :)

**Comment by fiddlycraft** [comment]
Yeah, there is just no way to actually update TT without people not being able to open IC2 Machines. They just don't work in most MV worlds.

**Comment by alicheek** [comment]
that's what nallar said if he did the server has a high chance of crashing and having issues.

**Comment by fiddlycraft** [comment]
I will just make a new world I guess and move everything to Overworld. Doesn't really bother me.

**Comment by alicheek** [comment]
I just informed my VIP players that their private world doesn't work with them atm most of them have main bases on the overworld any way so wasn't much of an issue

**Comment by LunNova** [comment]
Erm, this isn't permanently broken, will hopefully have it fixed soon.

**Comment by LunNova** [comment]
Not able to fix this without too much messing around, fixed in 1.5 but requires a forge client change, so it can't be backported to 1.4.

**Comment by dvernissi** [comment]
sorry to revisit this, but with the latest FTB releases I'm honestly considering in staying with mindcrack even further :P, can't accept redpower going away yet, so I would love to cling to 1.4.7 

would any of this be of any help? what about just a timer to fix the textures from time to time? 
or everytime someone does a multiverse teleport the task that ic2ckobfix.js runs? or something similar?
would it be too expensive to call: blockfix.net.updateTileEntityField if the IC2 mod is on?

this is what I got so far:
this piece of code, also known as the ic2ckobfix.js
lists the worlds, picks all loaded tile entities from it and applies a network update for texture facing on the IC2 object. on a schedule using ScriptCraftPlugin.
so the "ready" function creates scheduler part which calls for the 
"update" function which is the part that does most of the work

the problem is, it was made for 1.5.x with whatever obfuscated code it runs on.

oad(__folder + "../events/events.js");
plugin("blockfix", {
   update: function(){
      var players = server.getOnlinePlayers();

```
  for(var wid in blockfix.worlds){
     var te = blockfix.worlds[wid].field_73009_h;

     for (var ti=0;ti<te.size();ti++){

        if(te.get(ti).getFacing!=undefined){
           blockfix.net.updateTileEntityField(te.get(ti),"facing")
        }
     }
  }
```

   },
//    listWorldIds: function(){
//       for(var wi in blockfix.worlds){
         // var __world = blockfix.worlds[wi];
//          echo(__world.func_72912_H().func_76065_j()+" -> "+wi);
//       }
//    }
},false);

blockfix.cl = __plugin.getClass().getClassLoader().getParent();
blockfix.ic2 = blockfix.cl.loadClass("ic2.core.IC2").newInstance();
blockfix.net = blockfix.ic2.network;
blockfix.dmm = blockfix.cl.loadClass("net.minecraftforge.common.DimensionManager").newInstance();
blockfix.worlds = blockfix.dmm.getWorlds();

ready(function(){
   echo("[JS Cklob Temp Fix] Loading..");

   var fixMachine = new java.lang.Runnable({
         run: function(){
            blockfix.update();
         }
      });
   server.getScheduler().scheduleSyncRepeatingTask(__plugin,fixMachine,2_20,2_20);
});

the field field_73009_h, is known as loadedTileEntityList, it's a java ArrayList(), and in MCPC Legacy it's 'g'

so, the line: var te = blockfix.worlds[wid].field_73009_h
would end up being: var te = blockfix.worlds[wid].g

now this is all that we
